src/storage-types/src/sources/mysql.rs

```diff
@@ -312,19 +328,16 @@ impl RustType<ProtoMySqlSourceDetails> for MySqlSourceDetails {
 impl AlterCompatible for MySqlSourceDetails {
     fn alter_compatible(
         &self,
-        id: mz_repr::GlobalId,
-        other: &Self,
+        _id: mz_repr::GlobalId,
+        _other: &Self,
     ) -> Result<(), crate::controller::AlterError> {
-        if self.initial_gtid_set == other.initial_gtid_set {
-            Ok(())
-        } else {
-            tracing::warn!(
-                "MySqlSourceDetails incompatible at initial_gtid_set:\nself:\n{:#?}\n\nother\n{:#?}",
-                self,
-                other
-            );
-            Err(crate::controller::AlterError { id })
-        }
+        // TODO(roshan): We should verify here that the initial_gtid_set value for a given
+        // subsource remains the same. However, since a DROP SOURCE operation can drop a specific
+        // subsource without updating these `details`, and then an `ALTER SOURCE .. ADD SUBSOURCE`
+        // can re-add the same subsource with a new initial_gtid_set, there is no way for us to
+        // know that was a valid operation. We could consider updating `details` on DROP SOURCE
+        // or introducing a version or timestamp on each subsource to track changes.
```

Comment:
This is scary, and maybe hints towards having to store these initial gtid sets on the subsource object itself or use a more robust mechanism to control when we snapshot a source. What does the plan for fixing this look like?

Reply:
Yes I agree it's a little scary -- I had actually implemented a full comparison check here but had to remove it once I realized that the `details` object is not modified on a `DROP SOURCE` statement.

> store these initial gtid sets on the subsource object itself 

This is likely the plan that we'll need to implement. We'll need to add some sort of per-source-type metadata to the create-subsource-statements and `SourceExport` objects that can include this value on a per-subsource basis rather than per-table, to disambiguate a subsource being dropped and re-added for the same table.